circuits/spend.circom

```diff
@@ -12,36 +12,43 @@ include "./utils/assert.circom";
 include "./utils/convert.circom";
 include "./utils/public_commitment.circom";
 
-// Computes the encrypted balance (coin) using the Poseidon2 hash function 
-// with the given `balance` and `salt`. It then checks if the sum of 
-// `withdrawnBalance` and `remainingCoin` matches the encrypted balance.
+// Computes the encrypted balance (coin) using the Poseidon2 hash function
+// with the given `balance` and `salt`. Verifies that `withdrawnBalance` plus 
+// `remainingCoin` equals the encrypted balance, and includes both `fee` and 
+// `receiverAddress` in the public commitment for added flexibility.
 //
 // Example:
 //   balance:           1000
 //   burnKey:           123456
 //   coin:              Poseidon2(123456, 1000)
 //   withdrawnBalance:  200
 //   remainingCoin:     Poseidon2(123456, 800)
+//   fee:               50
+//   receiverAddress:   0x1234567890abcdef1234567890abcdef
 template Spend(maxAmountBytes) {
     signal input burnKey;
     signal input balance;
     signal input withdrawnBalance;
     signal input receiverAddress;
 
     signal output commitment;
+    signal input fee;
 
-    assert(maxAmountBytes <= 31); // To avoid field overflows
-
-    AssertGreaterEqThan(maxAmountBytes * 8)(balance, withdrawnBalance);
 
+    assert(maxAmountBytes <= 31); // To avoid field overflows
+    AssertBits(maxAmountBytes * 8)(fee);
+    AssertBits(maxAmountBytes * 8)(withdrawnBalance);
+    AssertBits(160)(receiverAddress);
+    AssertGreaterEqThan(maxAmountBytes * 8)(balance, withdrawnBalance + fee);
+    signal feeBytes[32] <== Num2BigEndianBytes(32)(fee);
```

Comment:
Put this where other `*Bytes* signals are created. (Use same order as PublicCommitment)